stdlib/result.gr

```diff
@@ -0,0 +1,63 @@
+export let isOk = (v) => {
+    match (v) {
+        | Ok(_)  => true
+        | _      => false
+    }
+};
+
+export let isErr = (v) => !isOk(v);
+
+export let toOption = (v) => {
+    match (v) {
+        | Ok(x)  => Some(x)
+        | _      => None
+    }
+}
+
+export let mapOk = (v, fn) => {
+    match (v) {
+        | Ok(x)  => Ok(fn(x))
+        | Err(e) => Err(e)
+    }
+}
+
+export let mapErr = (v, fn) => {
+    match (v) {
+        | Err(e) => Err(fn(v))
+        | _      => v
+    }
+}
+
+export let map = (v, fnOk, fnErr) => {
+    match (v) {
+        | Ok(x)  => fnOk(x)
+        | Err(x) => fnErr(x)
+    }
+}
+
+export let okOr = (v, default) => {
```

Comment:
In `Option`, this is `unwrapWithDefault` so maybe we should name this the same or `okWithDefault`?